Tools/AP_Periph/can.cpp

```diff
@@ -1277,6 +1302,41 @@ static void processRx(void)
     }
 }
 
+#ifdef HAL_PERIPH_CAN_MIRROR
+static void processMirror(void)
+{
+    const uint64_t deadline = AP_HAL::native_micros64() + 1000000;
+
+    for (auto &ins : instances) {
+        if (ins.iface == nullptr || ins.mirror_queue == nullptr) { // can't send on a null interface
+            continue;
+        }
+
+        for (uint32_t i = 0; i < ins.mirror_queue->available (); i++) { // limit how long we can loop
+            AP_HAL::CANFrame txmsg {};
+
+            if (!ins.mirror_queue->peek(txmsg)) {
+                break;
+            }
+
+            // FIXME: Do we need to set CANFD flags?
+            if (ins.iface->send(txmsg, deadline, 0) <= 0) {
```

Comment:
Doesn't this mess up things like source address?

Reply:
I'm pretty sure this is all fine, marking as resolved. If there's a reason this isn't then let me know.